src/Main.php

```diff
@@ -7,27 +7,38 @@
 use pocketmine\utils\TextFormat;
 
 class Main extends PluginBase {
-    public function onLoad() :void{
+
+	public function onLoad() :void{
         $this->saveResource("config.yml");
-        $config = $this->getConfig();
-        $log = $this->getLogger();
-        $version = "1.0.5";
-        if ($config->get("config-version") == $version){
-            return;
-        } else {
-            $log->notice("Your config is outdated!");
-            $log->info("Your old config.yml was as old-config.yml");
-            @rename($this->getDataFolder(). 'config.yml', 'old-config.yml');
-            $this->saveResource("config.yml");
-        }
+        $this->reloadConfig();
+        $this->checkConfig();
     }
     
-    
+    private function checkConfig() :void{
+	$log = $this->getLogger();
+	$pluginConfigResource = $this->getResource("config.yml");
+	$pluginConfig = yaml_parse(stream_get_contents($pluginConfigResource));
+	fclose($pluginConfigResource);
+	    $config = $this->getConfig();
+		
+	    if($pluginConfig == false) {
+	    	$log->critical("Invalid configuration.");
+	    	$this->getServer()->getPluginManager()->disablePlugin($this);
+	    	return;
+	    }
+
+	    if($config->get("config-version") === $pluginConfig["config-version"]) return;
+	    $log->notice("Your config is outdated!");
+	    $log->info("Your old config.yml is renamed as old-config.yml");
+	    @rename($this->getDataFolder(). 'config.yml', 'old-config.yml');
+	    $this->saveResource("config.yml");
+
+    }
+
 	public function onEnable() :void {
         $config = $this->getConfig();
         $log = $this->getLogger();
         $maxDistance = $config->get("max-distance");
```

Comment:
Config->get() used to return `false` when a key does not exist back in PM3. Does it no longer in PM4?

Reply:
```suggestion
        $maxDistance = $config->get("max-distance", null);
```
Maybe we should do this, just to be safe?

Reply:
> Maybe we should do this, just to be safe?

Perhaps change it to `"Invalid"` and change the logic to check for "Invalid" instead of isset?